lib/src/facade.dart

```diff
@@ -76,6 +76,20 @@ class Meili {
         distanceInMeters,
       );
 
+  //TODO(ahmednfwela): rework this method after Dart 3 lands with patterns

```

Comment:
If we adopt Dart 3, it will means that we will have to drop support for Dart 2, right?

Reply:
yes we will, but at the same time upgrading from dart 2 to dart 3 should be a no-cost operation, unless you are still using non null-safe code, which this package doesn't support anyway

Reply:
but we can leave that discussion until dart 3 lands, since only time will tell :)